ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -120,14 +120,11 @@ router bgp {{ router_bgp.as }}
 {%         endif %}
 {%     endfor %}
 {%     for neighbor_interface in router_bgp.neighbor_interfaces | arista.avd.natural_sort %}
-{%         set neighbor_interface_cli = "neighbor interface " ~ neighbor_interface %}
-{%         if router_bgp.neighbor_interfaces[neighbor_interface].peer_group is arista.avd.defined %}
-{%             set neighbor_interface_cli = neighbor_interface_cli ~ " peer-group " ~ router_bgp.neighbor_interfaces[neighbor_interface].peer_group %}
+{%         if router_bgp.neighbor_interfaces[neighbor_interface].peer_group is arista.avd.defined and router_bgp.neighbor_interfaces[neighbor_interface].remote_as is arista.avd.defined %}
```

Comment:
you are changing the logic here, so peer-group is no longer optional. Is that on prupose?
A simpler change to do what you want, was just to add and `elif` after the old line 128 and check for `peer_filter`.

Reply:
Nevermind. I can see that the old logic was not correct.
From EOS CLI:
`[no|default] neighbor interface INTFS peer-group PG ( ( remote-as ( AS_PLAIN_NUM | AS_DOT_NUM ) ) | ( peer-filter PEER_FILTER ) )`
`